File: 575.Distribute_Candies.py
class Solution(object):
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        
        type = set()
        for i in candies:
            type.add(i)
            
        return min(len(type),len(candies)/2)

File: 575.Distribute_Candies.java
import java.util.HashSet;

public class Solution {
    public int distributeCandies(int[] candies) {
        
        HashSet<Integer> hset = 
               new HashSet<Integer>();
        
        int len = candies.length;
        int max = len/2;
        for (int num:candies){
            hset.add(num);
        }
        
        return Math.min(hset.size(),max);
    }
}
